test/e2e/gameserver_test.go

```diff
@@ -818,20 +847,26 @@ func TestPlayerConnectAndDisconnect(t *testing.T) {
 	// deliberately do this before polling, to test the SDK returning the correct
 	// results before it is committed to the GameServer resource.
 	reply, err := e2eframework.SendGameServerUDP(gs, "PLAYER_CONNECTED 1")
-	assert.NoError(t, err)
+	if !assert.NoError(t, err) {
+		t.Fatalf("Could not message GameServer: %v", err)
+	}
 	assert.Equal(t, "true\n", reply)
 
 	reply, err = e2eframework.SendGameServerUDP(gs, "GET_PLAYERS")
-	assert.NoError(t, err)
+	if !assert.NoError(t, err) {
+		t.Fatalf("Could not message GameServer: %v", err)
+	}
 	assert.ElementsMatch(t, []string{"1", "2", "3"}, strings.Split(strings.TrimSpace(reply), ","))
 
 	reply, err = e2eframework.SendGameServerUDP(gs, "PLAYER_COUNT")
-	assert.NoError(t, err)
+	if !assert.NoError(t, err) {
+		t.Fatalf("Could not message GameServer: %v", err)
+	}
 	assert.Equal(t, "3\n", reply)
 
 	err = wait.Poll(time.Second, time.Minute, func() (bool, error) {
 		gs, err = framework.AgonesClient.AgonesV1().GameServers(framework.Namespace).Get(gs.ObjectMeta.Name, metav1.GetOptions{})
-		if err != nil {
+		if !assert.NoError(t, err) {
```

Comment:
Same as all the above - I don't think these are providing any kind of extra value, and could be confusing.

Reply:
There are places where we check errors with assert library in tests, that is why I decided to do the same:
https://github.com/googleforgames/agones/blob/master/pkg/util/apiserver/apiserver_test.go#L100
https://github.com/googleforgames/agones/blob/master/test/e2e/fleetautoscaler_test.go#L358
https://github.com/googleforgames/agones/blob/master/test/e2e/gameserverallocation_test.go#L356

As I see, currently there is no single approach and we use both options across the project. You are right, there is no big difference, but as far as this is a test file I decided to use a testing library here.
In which cases do you think we should use _err != nil_ or _assert.NoError()_ ?

Reply:
I don't know if I have strong opinions on either one. @roberthbailey I feel like you had opinions on this at one stage.

I feel like consolidating to:

```
	if !assert.NoError(t, err) {
		assert.FailNow(t, "the is my message for why this failed")
	}
```

Feels like the solution that best takes advantage of the test libraries we use, but that I think Robbie has preferred `if err != nil` more?

Thoughts?

Reply:
It's redundant to both assert.NoError and then call Fail, since the former calls fail already: https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1327

The one place where I don't recommend using the assert calls in tests is in goroutines, since calling asserts there don't necessarily bubble up nicely back to the main testing thread and give you a good idea of why a test failed. In those cases, doing "normal" error checking (`if err != nil`) and returning regular errors from the goroutines and then doing the asserts when they come back to the main routine is generally nicer. This is where I generally add comments about doing the normal error checking.

Reply:
> It's redundant to both assert.NoError and then call Fail, since the former calls fail already:

But `FailNow` will stop the test at that specific moment, which I think is the point. Or am I missing the point of your comment?

Reply:
Yes, aborting the test in case of an error is an idea, because if we go further nil value will be used. As a result, I’ve noticed nil pointer exceptions in some tests along with assert-fail messages in case if test fails.

Reply:
If we assert twice then we will duplicate error logs in the test output. So if we need to fail now I'm in favor of doing a regular `if err != nil` check to reduce noise in the test logs.